From stable-bounces@linux.kernel.org  Fri Jul 22 00:35:49 2005
Date: Fri, 22 Jul 2005 09:35:34 +0200
From: Patrick McHardy <kaber@trash.net>
To: "David S. Miller" <davem@davemloft.net>
Cc: Netfilter Development Mailinglist <netfilter-devel@lists.netfilter.org>,
        stable@kernel.org
Subject: [PATCH] [NETFILTER]: Fix deadlock in ip6_queue

[NETFILTER]: Fix deadlock in ip6_queue

Already fixed in ip_queue, ip6_queue was missed.

Signed-off-by: Patrick McHardy <kaber@trash.net>
Signed-off-by: Chris Wright <chrisw@osdl.org>
---

 net/ipv6/netfilter/ip6_queue.c |    2 ++
 1 files changed, 2 insertions(+)

Index: linux-2.6.12.y/net/ipv6/netfilter/ip6_queue.c
===================================================================
--- linux-2.6.12.y.orig/net/ipv6/netfilter/ip6_queue.c
+++ linux-2.6.12.y/net/ipv6/netfilter/ip6_queue.c
@@ -76,7 +76,9 @@ static DECLARE_MUTEX(ipqnl_sem);
 static void
 ipq_issue_verdict(struct ipq_queue_entry *entry, int verdict)
 {
+	local_bh_disable();
 	nf_reinject(entry->skb, entry->info, verdict);
+	local_bh_enable();
 	kfree(entry);
 }
 
